pilot/cmd/pilot-agent/status/ready/probe.go

```diff
@@ -101,6 +104,46 @@ func (p *Probe) checkServerState() error {
 	return nil
 }
 
+func (p *Probe) isEnvoyReady() error {
+	if se := p.checkServerState(); se != nil {
+		return se
+	}
+	if pe := p.pingVirtualListeners(); pe != nil {
```

Comment:
what about gateways?

Reply:
Looks like its implicitly skipped since GetVirtualListenerPorts will return `[]`. Could do with an explicitly check though?

Reply:
Yes. That is correct. It is implicitly skipped. I can add an explicit check to be sure.